### PR TITLE
Update owners file to reflect Juju/Charm knowledgable reviewers

### PR DESCRIPTION
**Pull request by marcoceppi**
**What this PR does / why we need it**:

Several reviewers have expressed confusion on why they're being added to pull requests related to cluster/juju. This is likely because the cluster OWNER wasn't updated from the original when created. This removes those who have expressed confusion to avoid noise in future updates to this directory.

Removed:
 - @pmorie [ref](https://github.com/kubernetes/kubernetes/pull/40814#issuecomment-282107925)
 - @jsafrane [ref](https://github.com/kubernetes/kubernetes/pull/42007#issuecomment-282246574)

Added:
 - @marcoceppi
 - @chuckbutler 
 - @mbruzek  

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by marcoceppi** [comment]
@k8s-bot gci gce e2e test this

File: cluster/juju/OWNERS [hunk 1]
```diff
@@ -4,13 +4,14 @@ approvers:
 - marcoceppi
 - mbruzek
 reviewers:
+- chuckbutler
+- marcoceppi
+- mbruzek
 - thockin
 - mikedanese
 - nikhiljindal
```
Comment (nikhiljindal) [inline cluster/juju/OWNERS]: Please feel free to remove me as well :)

File: cluster/juju/OWNERS [hunk 2]
```diff
@@ -4,13 +4,14 @@ approvers:
 - marcoceppi
 - mbruzek
 reviewers:
+- chuckbutler
```
Comment (nikhiljindal) [inline cluster/juju/OWNERS]: Dont you want to add castrojo as well?

Comment (mbruzek) [inline cluster/juju/OWNERS]: @nikhiljindal not for these directories. He helps us with documentation and I believe is a reviewer/approver on our docs directory.

**Pull request merged**